CenterForOpenScience/ember-osf-web
# [ENG-6181] Update search mirage view

-   Ticket: [ENG-6181]
-   Feature flag: n/a

## Purpose
- Update mirage view for trove index-card-search to return some more realistic results

## Summary of Changes
- Replace placekittens with placecats
- Remove hard-coded return object in mirage/views/search.ts
  - add a new factory-type object that creates resourceMetadata for index-cards of a given type

## Screenshot(s)
- search page with richer metadata fields!
Projects:
![image](https://github.com/user-attachments/assets/e518bb3e-c9ef-4eaf-92bc-9f2e392405de)

Registrations:
![image](https://github.com/user-attachments/assets/305386a2-377e-4460-9340-5da5add0ce3d)

Preprints:
![image](https://github.com/user-attachments/assets/02df90ae-9531-41a0-b343-9b3b35adcf6d)

Files:
![image](https://github.com/user-attachments/assets/aac1e70e-a309-4980-a9bd-748f7905a80f)

Users:
![image](https://github.com/user-attachments/assets/8138cd9c-9052-42b4-a20f-27b00655edf1)


## Side Effects
- None, really. The search page and search-result-card component should be a bit better testable now, but I don't know if I want to blow up the scope of this ticket just yet...

## QA Notes
- This is to update the info returned in mirage (our fake backend), so this should have no testable bit from QA team


[ENG-6181]: https://openscience.atlassian.net/browse/ENG-6181?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
